h/api/search/test/index_test.py

```diff
@@ -101,6 +112,245 @@ def log(self, patch):
+            mock.call(indexer, set(['id-1'])),
+        ]
+
+    def test_index_indexes_all_annotations_to_es(self, streaming_bulk):
+        ann_1, ann_2 = self.annotation(), self.annotation()
+
+        indexer = self.indexer(session=db.Session)
+        indexer.index()
+
+        streaming_bulk.assert_called_once_with(
+            indexer.es_client.conn, GeneratorEquals([ann_1, ann_2]),
+            chunk_size=mock.ANY, raise_on_error=False, expand_action_callback=mock.ANY)
+
+    def test_index_indexes_filtered_annotations_to_es(self, streaming_bulk):
+        ann_1, ann_2 = self.annotation(), self.annotation()
+
+        indexer = self.indexer(session=db.Session)
+        indexer.index([ann_2.id])
+
+        streaming_bulk.assert_called_once_with(
+            indexer.es_client.conn, GeneratorEquals([ann_2]),
+            chunk_size=mock.ANY, raise_on_error=False, expand_action_callback=mock.ANY)
+
+    def test_index_correctly_presents_bulk_actions(self, streaming_bulk, mock_request):
+        indexer = self.indexer(session=db.Session)
+        annotation = self.annotation()
+        results = []
+
+        def fake_streaming_bulk(*args, **kwargs):
+            ann = list(args[1])[0]
+            callback = kwargs.get('expand_action_callback')
+            results.append(callback(ann))
+            return set()
+
+        streaming_bulk.side_effect = fake_streaming_bulk
+
+        indexer.index()
+        assert results[0] == (
+            {'index': {'_type': indexer.es_client.t.annotation,
+                       '_index': indexer.es_client.index,
+                       '_id': annotation.id}},
+            presenters.AnnotationJSONPresenter(mock_request, annotation).asdict()
+        )
+
+    def test_index_returns_failed_bulk_actions(self, streaming_bulk):
+        indexer = self.indexer(session=db.Session)
+        ann_success_1, ann_success_2 = self.annotation(), self.annotation()
+        ann_fail_1, ann_fail_2 = self.annotation(), self.annotation()
+
+        def fake_streaming_bulk(*args, **kwargs):
+            for ann in args[1]:
+                if ann.id in [ann_fail_1.id, ann_fail_2.id]:
+                    yield (False, {'index': {'_id': ann.id}})
+                elif ann.id in [ann_success_1.id, ann_success_2.id]:
+                    yield (True, {'index': {'_id': ann.id}})
+
+        streaming_bulk.side_effect = fake_streaming_bulk
+
+        result = indexer.index()
+        assert result == set([ann_fail_1.id, ann_fail_2.id])
+
+    @pytest.fixture
+    def indexer(self, session=None):
+        if session is None:
+            session = mock.MagicMock()
+        return index.BatchIndexer(session, mock.MagicMock(), DummyRequest())
+
+    @pytest.fixture
+    def index(self, patch):
+        return patch('h.api.search.index.BatchIndexer.index')
+
+    @pytest.fixture
+    def streaming_bulk(self, patch):
+        return patch('h.api.search.index.es_helpers.streaming_bulk')
+
+    @pytest.fixture
```

Comment:
Doesn't look like you're using it as a fixture any more so this should probably go.